pyrt/renderer/simplert.py

```diff
@@ -69,22 +69,39 @@ def _shadow(self, scene: Scene, hitrecord: HitRecord) -> tuple:
             fs /= 4.
 
         return fs,local_num_shadow_rays
+    
+    def _recurse_shade(self, scene: Scene, ray: Ray, iteration_num: int) -> tuple:
+        hitrecord = HitRecord()
+        hit, r, g, b = self._shade(scene, ray, hitrecord)
 
-    def _reflect(self, r: int, g: int, b: int, scene: Scene, ray: Ray, hitrecord: HitRecord) -> tuple:
-        reflect_ray = Ray(hitrecord.point, reflect3(hitrecord.normal_g, ray.direction))
-        reflect_hitrecord = HitRecord()
-
-        hit, rnew, gnew, bnew = self._shade(scene, reflect_ray, reflect_hitrecord)
+        if not hit or iteration_num == 0:
+            return r, g, b
+        
+        if hitrecord.material.reflectivity != 0.0:
```

Comment:
Should be a fuzzy EPSILON comparison

Be careful when using floating-point numbers

-EPS < a < EPS